src/openforms/submissions/models/submission.py

```diff
@@ -727,3 +737,25 @@ def waiting_on_cosign(self) -> bool:
             return True
 
         return False
+
+    _registration_backend: str | FormRegistrationBackend | None = None
+
+    @property
+    def registration_backend(self) -> FormRegistrationBackend:
+        if self.finalised_registration_backend:
+            return self.finalised_registration_backend
+
+        match self._registration_backend:
+            case str(key):
+                self._registration_backend = self.form.registration_backends.get(
+                    key=key
+                )
+            case None:
+                self._registration_backend = self.form.registration_backends.first()
```

Comment:
No even for new cases: if I specify just one backend, no logic rules are needed.

Forgetting to set one or setting a wrong one was specced as undefined behaviour.

I think the correct behaviour would be to skip registration. But as per your other comment, I'll revert to the first in a defined order, and also log debug info to the timeline of the submission.

Reply:
this was mostly talking about existing data - it seems weird that you can have:

1. an existing submission that only has max 1 registration backend
2. you update the form for that submission and add another registration backend
3. looking at the existing submission now suddenly points to the new registration backend, which would mess up registration retries

So using defined order here should now be fine if the order lines up with the data migration that copies form registration configuration into the new model. But I think it'd have removed this implicit behaviour and recorded the registration backend used explicitly on the submission model after the new field was/is added (and I think in particular if we switch to a CharField instead of FK this is easily achieved, even more so if we add the charfield to the submission first, copy the information over and only then update the `Form` schema :thinking: , this would remove all ambiguity)